src/lib/wallet/GoodWalletClass.js

```diff
@@ -477,19 +477,30 @@ export class GoodWallet {
     const withdrawHash = '0x39ca68a9f5d8038e871ef25a6622a56579cda4a6eedf63813574d23652e94048'
     const cancelHash = '0xb1f6a8f6b8fb527cfeec0df589160bc2a92ff715097117cb250e823c7106bc2f'
 
-    const otpPromise = otpAddress
-      ? Promise.all([
-          API.getOTPLEvents(account, networkId, otpAddress, startBlock, currentBlock, withdrawHash),
-          API.getOTPLEvents(account, networkId, otpAddress, startBlock, currentBlock, cancelHash),
-        ])
-      : Promise.resolve()
-
-    ;[results, otpResults] = await Promise.all([tokenPromise, otpPromise])
-
-    await Promise.all([
-      results?.length > 0 ? this.processEvents(results, startBlock) : undefined,
-      otpResults?.length > 0 ? this.processEvents(flatten(otpResults), startBlock) : undefined,
-    ])
+    try {
+      const otpPromise = otpAddress
+        ? Promise.all([
+            API.getOTPLEvents(account, networkId, otpAddress, startBlock, currentBlock, withdrawHash),
+            API.getOTPLEvents(account, networkId, otpAddress, startBlock, currentBlock, cancelHash),
+          ])
+        : Promise.resolve()
+
+      ;[results, otpResults] = await Promise.all([tokenPromise, otpPromise])
+
+      await Promise.all([
+        results?.length > 0 ? this.processEvents(results, startBlock) : undefined,
+        otpResults?.length > 0 ? this.processEvents(flatten(otpResults), startBlock) : undefined,
+      ])
+    } catch (error) {
```

Comment:
@johnsmith-gooddollar @sirpy 
I now added a try/catch block here to log an error for the isArray safe-guard.
having this try/catch here means it will still finalize the syncTxFromExplorer and update lastblock accordingly.

Should we already log here? or do we push it all to the outer-catch block of syncTxFromExplorer caller and run the flow for syncTxFromBlockchain?

Reply:
2. Aren't they both catched because of the Promise.all on line 488? Where we await the resolution of both tokenPromise/otpPromise?

Reply:
Promise all executes them simultaenously therefore more chances to hit rate limit. if you do
await getOTPL
await getOTPL
there's less chance to hit rate limits.

Reply:
I understand that, I was referring specifically to your question 'where is the exception handling for getTransactionsFromexplorer' and I say: its in the
```
Promise.all([tokenPromise <-- getTxFromEpxlorer, otpPromise <-- getOTPevents])
```

Reply:
```javascript
let withdrawEvents = []
let cancelEvents = []

const getOTPL = hash => API.getOTPLEvents(account, networkId, otpAddress, startBlock, currentBlock, hash)

const tokenEvents = await API.getTokenTxs(tokenAddress, account, networkId, startBlock).then(results =>
  results.map(result => ({ ...result, transactionHash: result.transactionHash || result.hash })),
)

await this.processEvents(tokenEvents)

if (otpAddress) {
  withdrawEvents = await getOTPL(withdrawHash)
  cancelEvents = await getOTPL(cancelHash)
}

// will parallel it all together, see fn implementation
await this.processEvents([
   ...tokenEvents,
  ...withdrawEvents,
  ...cancelEvents, 
])
```